libgobuster/dir.go

```diff
@@ -59,6 +61,22 @@ func MakeRequest(s *State, fullUrl, cookie string) (*int, *int64) {
 		req.Header.Set("User-Agent", s.UserAgent)
 	}
 
+	if s.RandomUserAgent && s.UserAgent == "" {
+		b, err := ioutil.ReadFile("user-agents.txt")
+		if err != nil {
+			fmt.Print(err)
```

Comment:
You catch the error here, print it, but then execution will just continue I believe? We would probably be better returning the error here, or (maybe, but it's gross) panic'ing?

Reply:
How do you suggest to return if error then?

`return nil, nil` ?

Reply:
Normally it would be ‘return nil, er’ or similar. On my phone so can’t check the return type, may need to add a 3rd type (error) if it’s already returning 2. Then return nil, nil, error.

Can’t remember if I started implementing multierror in this or not, if so, possibly using that type might be beneficial.

Would also have to add an error check around anywhere that calls this.

Ideally we would be bubbling any errors back up to the highest point, so it can decide how to handle them (or die). Rather than the current panics we have scattered around the place.

Reply:
A little divergent from initial PR, but implementing that pattern for this function would help to clear up some of the existing tech debt.

Reply:
Function MakeRequest returns `(*int, *int64)` so it gives an error if you try to `return nil, err`.

I'll leave it as the rest of the returns in that function, as `nil, nil`